### PR TITLE
[Supplier] Add `MsgUpdateParam` to application module

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

```bash
ignite scaffold message update-param --module supplier --signer authority name as_type --response params
```

Adds the `MsgUpdateParam` message so that the supplier module may update individual parameters following section 0 from the updated docs (see #839).

## Dependencies

- #809
- #843 
- #844 
- #845
- #847
- #848

## Dependents
  
- #850
- #857
- #852 
- #861 
- #851 
- #863 

## Issue

- #612

## Type of change

Select one or more from the following:

- [x] New feature, functionality or library
- [ ] Consensus breaking; add the `consensus-breaking` label if so. See #791 for details
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Documentation**: `make docusaurus_start`; only needed if you make doc changes
- [x] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- [ ] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [x] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [x] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: proto/poktroll/supplier/tx.proto [hunk 1]
```diff
@@ -9,33 +10,32 @@ import "cosmos/msg/v1/msg.proto";
...
+  option           (amino.name) = "poktroll/x/supplier/MsgUpdateParams";
+  
   // authority is the address that controls the module (defaults to x/gov unless overwritten).
   string authority = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
 
   // TODO_IMPROVE(#322): The requirement to provide all params is adopted from the
   // latest Cosmos SDK version. We should look into either improving this ourselves
   // or seeing if it is on their roadmap.
-
+
```
Comment (red-0ne) [inline proto/poktroll/supplier/tx.proto:38]:
```suggestion

```

File: proto/poktroll/supplier/tx.proto [hunk 2]
```diff
@@ -9,33 +10,32 @@ import "cosmos/msg/v1/msg.proto";
...
+  rpc UpdateParam     (MsgUpdateParam    ) returns (MsgUpdateParamResponse    );
 }
 // MsgUpdateParams is the Msg/UpdateParams request type.
 message MsgUpdateParams {
-  option (cosmos.msg.v1.signer) = "authority";
-  option (amino.name) = "poktroll/x/supplier/MsgUpdateParams";
-
+  option (cosmos.msg.v1.signer) =                           "authority";
+  option           (amino.name) = "poktroll/x/supplier/MsgUpdateParams";
+
```
Comment (red-0ne) [inline proto/poktroll/supplier/tx.proto:31]:
```suggestion

```

**Review by red-0ne — APPROVED** [review]
Tiny NIT but LGTM!

File: proto/poktroll/supplier/tx.proto (hunk 2 above)
Comment (bryanchriswhite) [reply proto/poktroll/supplier/tx.proto:31]: My personal preference is to have the `option`s separate from fields by one blank line. I have not surveyed the proto messages to see what's more consistent nor am I aware of any particular convention about this. Do you have a strong opinion or a rationale you could chare?

File: proto/poktroll/supplier/tx.proto (hunk 1 above)
Comment (bryanchriswhite) [reply proto/poktroll/supplier/tx.proto]: I think this comment is actually resolved at this point. Removing.

**Pull request merged**